client/galaxy/scripts/layout/data.js

```diff
@@ -16,4 +16,42 @@ export default class Data {
             }
         }).$mount(vm);
     }
+
+    /**
+     * Creates a history dataset by submitting an upload request
+     */
+    create(options) {
+        let history_panel = Galaxy.currHistoryPanel;
+        let history_id = options.history_id;
+        if (!history_id && history_panel) {
+            history_id = history_panel.model.get("id");
+        }
+        $.uploadpost({
+            url: `${Galaxy.root}api/tools`,
+            success: (response) => {
+                if (history_panel) {
+                    history_panel.refreshContents();
+                }
+                if (options.success) {
+                    options.success(response);
+                }
+            },
+            error: options.error,
+            data: {
+                payload: {
+                    tool_id: "upload1",
```

Comment:
could this be done with the fetch tool instead ? Isn't that intended to replace upload1 in the future ?

Reply:
Interesting, I have not used that tool yet in that fashion. Do we have an example? It should be straightforward to redirect the post if needed, either now or in the future.

Reply:
@mvdbeek I would prefer to keep using the current upload tool until the new tool is fully functional and ready to replace it.